File: ReverseNumber.cpp
	using namespace std;
	#include<iostream>
	#include<stdio.h>

	int getDigits(int);
	signed int reverseNumber(int);

	int main(){
		int num, digits;
		cout<<"Input any number: ";
		cin>>num;
		digits= getDigits(num);
		cout<<"\nNo of digits in "<<num<<"="<<digits;
		signed int reverse = reverseNumber(num);
		cout<<"\nReverse of "<<num<<"="<<reverse;

	return 0;	
	}
int getDigits(int num){
	int count=0;
	while(num!=0){
			num/=10; //Successively divides the number to get number of digits
			count++; 
	}
	return count;
}

signed int reverseNumber(int num){
	signed int rev=0;
	while(num!=0){
		rev=rev+num%10;
		num=num/10;
		cout<<"\nrev="<<rev;
		rev=rev*10;
	}
	return (rev/10);
	// last multiplication needs to be deducted in order to remove the excess multiplication
}

/*
num=456
digits=3
rev=num%10=6
rev=

*/

File: ReverseNumber.java
// javac ReverseNumber.java
// java ReverseNumber

import java.util.Scanner;

public class ReverseNumber {

    static int num; 
    static int digits;
//mandatory to declare num and digits as static because in main we are passing them as args and if they're not static, they'll not be available in main function. 
    ReverseNumber(){
        Scanner inputObj= new Scanner(System.in);
        System.out.println("Please input any number: ");
        num=inputObj.nextInt();
        digits=0;
    }

    private int getDigits(int num) {
        int count=0;
        while(num!=0){
            num=num/10;
            count++;
        }
        return count;
    }
    private int reverseNumber(int num) {
        int rev=0;
        while(num!=0){
            rev=rev+num%10;
            num=num/10;
            rev=rev*10;
        }    
    return (rev/10);
    }

    public static void main(String[] args) {
        System.out.println("Hello");
        ReverseNumber obj = new ReverseNumber();
        System.out.println(obj.getDigits(num));
        System.out.println(obj.reverseNumber(num));

    }
}